modules/apps/transfer/types/token.go

```diff
@@ -0,0 +1,61 @@
+package types
+
+import (
+	"strings"
+
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	denominternal "github.com/cosmos/ibc-go/v8/modules/apps/transfer/internal/denom"
+)
+
+// Validate validates a token denomination and trace identifiers.
+func (t Token) Validate() error {
+	if err := sdk.ValidateDenom(t.Denom); err != nil {
+		return errorsmod.Wrap(ErrInvalidDenomForTransfer, err.Error())
+	}
+
+	if len(t.Trace) == 0 {
+		return nil
+	}
+
+	trace := strings.Join(t.Trace, "/")
+	identifiers := strings.Split(trace, "/")
```

Comment:
am I missing something? This looks redundant? I guess it is taking the "portid/channelid" pair and turning it into "portid", "channelid", might be nice to change the validate trace identifiers to not need to do this, or to add a comment explaining this logic since it looks odd

Reply:
pretty sure this is redundant ad we can just pass `t.Trace` directly as is :thinking:

Reply:
unfortunately not, the []string types used are of different values. 

```
t.Trace = []string{"portid/channelid", "portid/channelid"}
identifiers = []string{"portid", "channelid", "portid", "channelid"}
```

It seems there is agreement on https://github.com/cosmos/ibc-go/issues/6362, which should remove this issue